### PR TITLE
fix: coordinates formatting in bbox2wktpolygon

### LINKED ISSUE #53: Incorrect formatting of coordinates in bbox2wktpolygon function
# Description
The `bbox2wktpolygon` function in the `util.py` file uses `%.2f` for formatting coordinates in the resulting WKT Polygon. This formatting limits the number of decimal places in the output, leading to inaccurate representation of coordinates.

# Environment

- operating system:
- Python version: 3.8
- pycsw version: v3.4.8
- source/distribution
  - [ ] git clone
  - [ ] DebianGIS/UbuntuGIS
  - [ ] PyPI
  - [ ] zip/tar.gz
  - [ ] other (please specify):
- web server
  - [ ] Apache/mod_wsgi
  - [ ] CGI
  - [ ] other (please specify): 

# Steps to Reproduce
1. Use the `bbox2wktpolygon` function with a bounding box input.
2. Observe the resulting Well-Known Text (WKT) Polygon and note the limited precision in the coordinates.

# Additional Information
**Expected Behavior:**
The `bbox2wktpolygon` function should represent coordinates with full precision, providing accurate WKT Polygons.

**Proposed Fix:**
Modify the formatting in the `return` statement to use `%s` instead of `%.2f` for coordinates.

```python
return 'POLYGON((%s %s, %s %s, %s %s, %s %s, %s %s))' \
    % (minx, miny, minx, maxy, maxx, maxy, maxx, miny, minx, miny)

### PR DESCRIPTION
**Pull request by netanelC**
# Overview
Fixes the issue of incorrect formatting of coordinates in the `bbox2wktpolygon` function by updating the coordinate formatting in the return statement.

# Related Issue / Discussion
Closes #53 

# Additional Information
- This change modifies the `bbox2wktpolygon` function in `pycsw/core/util.py` to use an environment variable with the desired bbox precise digits instead of `%.2f` for coordinate formatting (in case the env is not provided, the default value will be 2 as source).
- Coordinates are now restricting the number of decimal places according to client requirements.

# Contributions and Licensing

(as per https://github.com/geopython/pycsw/blob/master/CONTRIBUTING.rst#contributions-and-licensing)

- [x] I'd like to contribute [feature X|bugfix Y|docs|something else] to pycsw. I confirm that my contributions to pycsw will be compatible with the pycsw license guidelines at the time of contribution.
- [x] I have already previously agreed to the pycsw Contributions and Licensing Guidelines

### CODE REVIEW

**Review by asafMasa — APPROVED** [review]

**Comment by RonitKissis** [comment]
great job but can it be forwarded by env variable? make it configurable so that the default willl be %.2f if no variable was given?

**Review by RonitKissis — CHANGES_REQUESTED** [review]
great job but can it be forwarded by env variable? make it configurable so that the default willl be %.2f if no variable was given?

File: pycsw/core/util.py [hunk 1]
```diff
@@ -220,10 +220,39 @@ def bbox2wktpolygon(bbox):
...
 
+def format_coord(number):
+    """Return desired precision of coordinate
+    Uses the env "COORDINATE_PRECISION" to set the precise number of digits after dot (for desired accurate result).
+    If the env is not specified, the default precise number is 2 digits after the dot.
+
+    Parameters
+    ----------
+    number: float
+        A number presenting part of the coordinate
```
Comment (shimoncohen) [inline pycsw/core/util.py:236]: `presenting` -> `representing`.
Even better (IMO):
`A number presenting part of the coordinate` -> `A coordinate single value`

File: pycsw/core/util.py [hunk 2]
```diff
@@ -220,10 +220,39 @@ def bbox2wktpolygon(bbox):
...
+
+    Returns
+    -------
+    float
+        The number with desired formatted precision
```
Comment (shimoncohen) [inline pycsw/core/util.py:241]: `The number with desired formatted precision` -> `The number after formatting`.

File: pycsw/core/util.py [hunk 3]
```diff
@@ -220,10 +220,39 @@ def bbox2wktpolygon(bbox):
...
+
+    """
+
+    precision = int(os.environ.get('COORDINATE_PRECISION', 2))
+    if(str(number).find('.') == -1):
+        return number
+    abs_number = abs(number)
+    num_digits_before_dot = str(abs_number)[::1].find('.')
```
Comment (shimoncohen) [inline pycsw/core/util.py:249]: Default step is 1, why set it explicitly?
And string has [`find`](https://www.w3schools.com/python/ref_string_find.asp) function, why not use it directly?

File: pycsw/core/util.py [hunk 4]
```diff
@@ -220,10 +220,39 @@ def bbox2wktpolygon(bbox):
...
+    if(abs_number<1):
```
Comment (shimoncohen) [inline pycsw/core/util.py:250]: Please use spaces in your code when possible for readability.
This is true for other places in this file too.

File: pycsw/core/util.py [hunk 5]
```diff
@@ -220,10 +220,39 @@ def bbox2wktpolygon(bbox):
...
+        num_digits_before_dot=0
+    formatted_number = float(('{:.' + str(int(precision) + num_digits_before_dot) + '}').format(abs_number))
```
Comment (shimoncohen) [inline pycsw/core/util.py:252]: `precision` is already a number, no need for casting.

Comment (shimoncohen) [inline pycsw/core/util.py:252]: Why aren't you formatting in the form `%.{precision}f`?

File: pycsw/core/util.py (hunk 3 above)
Comment (shimoncohen) [inline pycsw/core/util.py:249]: Why not move this above the `if(str(number).find('.') == -1)` and change the condition?

File: pycsw/core/util.py [hunk 6]
```diff
@@ -220,10 +220,39 @@ def bbox2wktpolygon(bbox):
...
+    if(number<0):
```
Comment (shimoncohen) [inline pycsw/core/util.py:253]: Maybe use [ternary operator](https://www.geeksforgeeks.org/ternary-operator-in-python/).

File: pycsw/core/util.py [hunk 7]
```diff
@@ -220,10 +220,39 @@ def bbox2wktpolygon(bbox):
...
     if bbox.startswith('ENVELOPE'):
         bbox = wktenvelope2bbox(bbox)
-    minx, miny, maxx, maxy = [float(coord) for coord in bbox.split(",")]
-    return 'POLYGON((%.2f %.2f, %.2f %.2f, %.2f %.2f, %.2f %.2f, %.2f %.2f))' \
+    minx, miny, maxx, maxy = [format_coord(float(coord)) for coord in bbox.split(",")]
+    wktGeometry = 'POLYGON((%s %s, %s %s, %s %s, %s %s, %s %s))' \
         % (minx, miny, minx, maxy, maxx, maxy, maxx, miny, minx, miny)
+    return wktGeometry
 
+def format_coord(number):
```
Comment (shimoncohen) [inline pycsw/core/util.py:228]: I think this function can be replaced by `f"{number:.{precision}f}""`. This supports negative values too.

**Review by shimoncohen — CHANGES_REQUESTED** [review]

**Review by RonitKissis — APPROVED** [review]

**Pull request merged**